### ISSUE TITLE
Tracer of the removed fluid in hydro module will lose information in a restart simulation

### ISSUE THREAD

**Issue by piyueh** [barbagroup/geoclaw#15]
If a simulation is a restart session, then the tracer will lose the information from the previous simulation. But given that land-spill simulations are usually very short, restarting does not required frequently. So this issue will not be addressed in the near future.

**Comment by mandli** [comment]
One thing of note, if the tracer value is stored in the q array then it will be available.  The entirety of the `alloc` array, which stores the grid data, aux data, etc., all gets saved (hence the size of the restart files).

**Comment by piyueh** [comment]
Thank you @mandli. The reason I didn't use `q` array for the tracer is that only cells surrounding water bodies have values in the tracer. Using `q` array for this purpose will waste memory on cells that will never have non-zero values.

**Comment by mandli** [comment]
Ah, so you are storing things in a very small part of the domain yourself?

**Comment by piyueh** [comment]
Not really. The tracer is a fixed Cartesian mesh (or says, a raster in GIS terminology) covering the whole computational domain. But the data on this "tracer mesh" is not a dense array, unlike `q` or `aux`. The data is a sparse matrix in CSR format. And the locations of non-zero entries in this sparse matrix are the cells surrounding water bodies. So, no matter how many cells are in this tracer mesh in total, the memory consumed is only relevant to the number of cells surrounding water bodies.

**Comment by mandli** [comment]
Now I understand, this is not actually for the oil itself but the oil in the water.

**Comment by piyueh** [comment]
Yes, you're right!
